modules/appeals_api/app/models/concerns/appeals_api/central_mail_status.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+require 'central_mail/service'
+
+module AppealsApi
+  module CentralMailStatus
+    extend ActiveSupport::Concern
+
+    STATUSES = %w[pending submitting submitted processing error uploaded received success expired].freeze
+
+    CENTRAL_MAIL_STATUS_TO_APPEALS_ATTRIBUTES = lambda do
+      hash = Hash.new { |_, _| raise ArgumentError, 'Unknown Central Mail status' }
+      hash['Received'] = { status: 'received' }
+      hash['In Process'] = { status: 'processing' }
+      hash['Processing Success'] = hash['In Process']
+      hash['Success'] = { status: 'success' }
+      hash['Error'] = { status: 'error', code: 'DOC202' }
+      hash['Processing Error'] = hash['Error']
+      hash
+    end.call.freeze
+    # ensure that statuses in map are valid statuses
+    raise unless CENTRAL_MAIL_STATUS_TO_APPEALS_ATTRIBUTES.values.all? do |attributes|
+      [:status, 'status'].all? do |status|
+        !attributes.key?(status) || attributes[status].in?(STATUSES)
+      end
+    end
+
+    CENTRAL_MAIL_ERROR_STATUSES = ['Error', 'Processing Error'].freeze
+    raise unless CENTRAL_MAIL_ERROR_STATUSES - CENTRAL_MAIL_STATUS_TO_APPEALS_ATTRIBUTES.keys == []
+
+    RECEIVED_OR_PROCESSING = %w[received processing].freeze
+    raise unless RECEIVED_OR_PROCESSING - STATUSES == []
+
+    COMPLETE_STATUSES = %w[success error].freeze
+    raise unless COMPLETE_STATUSES - STATUSES == []
```

Comment:
I realize that you're just copy and pasting the pre-existing code, but...

I'm not really fond using `raise` to verify that all these "sub" statuses are contained in the "parent" status object.

Could we do something along these lines?

```
RECEIVED_OR_PROCESSING = %w[received processing].freeze
COMPLETE_STATUSES = %w[success error].freeze

STATUSES = [*RECEIVED_OR_PROCESSING, *COMPLETE_STATUSES].freeze
```

I feel like this would convey out intent better that all the "sub" statuses also need to be part of the "parent" status object.

This could always be another ticket too...

Reply:
^ Same. I think we should create a new ticket and discuss.